Time series task

EgorKraevTransferwise opened the pull request:
Merge all the time series refinements into the new structure
<!-- Please give a short summary of the change and the problem this solves. -->

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

markharley committed:
Refactor into automl subpackage

Moved some of the packages into an automl subpackage to tidy before the
task-based refactor. This is in response to discussions with the group
and a comment on the first task-based PR.

Only changes here are moving subpackages and modules into the new
automl, fixing imports to work with this structure and fixing some
dependencies in setup.py.

markharley committed:
Fix doc building post automl subpackage refactor

markharley committed:
Fix broken links in website post automl subpackage refactor

markharley committed:
Fix broken links in website post automl subpackage refactor

markharley committed:
Remove vw from test deps as this is breaking the build

markharley committed:
WIP

markharley committed:
Move default back to the top-level

I'd moved this to automl as that's where it's used internally, but had
missed that this is actually part of the public interface so makes sense
to live where it was.

markharley committed:
Re-add top level modules with deprecation warnings

flaml.data, flaml.ml and flaml.model are re-added to the top level,
being re-exported from flaml.automl for backwards compatability. Adding
a deprecation warning so that we can have a planned removal later.

qingyun-wu committed:
Merge branch 'main' into subpackage-refactor-for-automl

markharley committed:
WIP - Notes below

Got to the point where the methods from AutoML are pulled to
GenericTask. Started removing private markers and removing the passing
of automl to these methods. Done with decide_split_type, started on
prepare_data. Need to do the others after

qingyun-wu committed:
Merge branch 'main' into subpackage-refactor-for-automl

qingyun-wu committed:
Merge branch 'main' into subpackage-refactor-for-automl

markharley committed:
Merge remote-tracking branch 'upstream/main' into subpackage-refactor-for-automl

EgorKraevTransferwise committed:
Merge microsoft/main into here

markharley committed:
Merge remote-tracking branch 'upstream/main' into subpackage-refactor-for-automl

markharley committed:
Fix model.py line-endings

markharley committed:
Merge remote-tracking branch 'upstream/main' into extract-task-class-from-automl

markharley committed:
Re-add generic_task

markharley committed:
Merge remote-tracking branch 'upstream/main' into extract-task-class-from-automl

EgorKraevTransferwise committed:
Most of the merge done, test_forecast_automl fit succeeds, fails at predict()

EgorKraevTransferwise committed:
Remaining fixes - test_forecast.py passes

EgorKraevTransferwise committed:
Comment out holidays-related code as it's not currently used

EgorKraevTransferwise committed:
Further holidays cleanup

EgorKraevTransferwise committed:
Fix imports in a test

EgorKraevTransferwise committed:
tidy up validate_data in time series task

EgorKraevTransferwise committed:
Test fixes

EgorKraevTransferwise committed:
Fix tests: add Task.__str__

EgorKraevTransferwise committed:
Fix tests: test for ray.ObjectRef

EgorKraevTransferwise committed:
Hotwire TS_Sklearn wrapper to fix test fail

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/extract-task-class-from-automl' into multivariate_target

# Conflicts:
#	flaml/automl/model.py
#	flaml/default/suggest.py

EgorKraevTransferwise committed:
Attempt at test fix

EgorKraevTransferwise committed:
Fix test where val_pred_y is a list

EgorKraevTransferwise committed:
Attempt to fix remaining tests

EgorKraevTransferwise committed:
Push to retrigger tests

EgorKraevTransferwise committed:
Push to retrigger tests

EgorKraevTransferwise committed:
Push to retrigger tests

EgorKraevTransferwise committed:
Push to retrigger tests

EgorKraevTransferwise committed:
Remove plots from automl/test_forecast

markharley committed:
Merge remote-tracking branch 'upstream/main' into extract-task-class-from-automl

markharley committed:
Merge remote-tracking branch 'origin/extract-task-class-from-automl' into extract-task-class-from-automl

markharley committed:
Remove unused data size field from Task

markharley committed:
Merge remote-tracking branch 'upstream/main' into extract-task-class-from-automl

markharley committed:
Fix import for CLASSIFICATION in notebook

EgorKraevTransferwise committed:
Monkey patch TFT to avoid plotting, to fix tests on MacOS

EgorKraevTransferwise committed:
Monkey patch TFT to avoid plotting v2, to fix tests on MacOS

EgorKraevTransferwise committed:
Monkey patch TFT to avoid plotting v2, to fix tests on MacOS

EgorKraevTransferwise committed:
Fix circular import

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/extract-task-class-from-automl' into multivariate_target

# Conflicts:
#	flaml/automl/state.py

EgorKraevTransferwise committed:
remove redundant code in task.py post-merge

ZmeiGorynych commented on flaml/automl/time_series/tft.py:
```diff
@@ -0,0 +1,193 @@
+        from pytorch_lightning.callbacks import EarlyStopping, LearningRateMonitor
+        from pytorch_lightning.loggers import TensorBoardLogger
+
+        # a bit of monkey patching to fix the MacOS test
```
As all this function appeared to do is some plotting during a fit (which we don't do for other models) and somehow caused a fail on MacOS, with no way I could find to turn it off,  I monkey-patch it here to do nothing instead :)

int-chaos replied:
There is actually a way to turn it off. See https://github.com/microsoft/FLAML/blob/cdc4201188565e74507c026f86abe540ade4cb72/flaml/automl/model.py#L2275

``` python
        def _fit(log):
            default_trainer_kwargs = dict(
                gpus=kwargs.get("gpu_per_trial", [0])
                if torch.cuda.is_available()
                else None,
                max_epochs=max_epochs,
                gradient_clip_val=gradient_clip_val,
                callbacks=[LearningRateMonitor(), early_stop_callback]
                if log
                else [early_stop_callback],
                logger=log,
            )
            trainer = pl.Trainer(
                **default_trainer_kwargs,
            )
            tft = TemporalFusionTransformer.from_dataset(
                training,
                **params,
                lstm_layers=2,  # 2 is mostly optimal according to documentation
                output_size=7,  # 7 quantiles by default
                loss=QuantileLoss(),
                log_interval=10 if log else 0,
                # uncomment for learning rate finder and otherwise, e.g. to 10 for logging every 10 batches
                reduce_on_plateau_patience=4,
            )
            # fit network
            trainer.fit(
                tft,
                train_dataloaders=train_dataloader,
                val_dataloaders=val_dataloader,
            )
            return trainer
        trainer = _fit(log=False)
```

EgorKraevTransferwise committed:
Fix test: set svd_solver="full" in PCA

markharley committed:
Update flaml/automl/data.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

markharley committed:
Fix review comments

markharley committed:
Fix task -> str in custom learner constructor

markharley committed:
Merge remote-tracking branch 'upstream/main' into extract-task-class-from-automl

markharley committed:
Remove unused CLASSIFICATION imports

int-chaos commented on flaml/automl/time_series/ts_model.py:
```diff
@@ -0,0 +1,694 @@
+
+
+class TS_SKLearn(TimeSeriesEstimator):
+    """The class for tuning SKLearn Regressors for time-series forecasting, using hcrystalball"""
```
should remove "using hcrystalball" if we're not using it anymore

EgorKraevTransferwise replied:
Good point, will do

int-chaos commented on flaml/automl/time_series/smooth.py:
```diff
@@ -0,0 +1,106 @@
+    I = np.eye(15)
+    step = 5
+    points = 38
+    out = moving_window_smooth(I, step, True)
```
I don't see smooth.py being used anywhere. Is there a reason to include this in this pr?

EgorKraevTransferwise replied:
You're right, it's used in the multiscale model which isn't part of this PR - will remove

EgorKraevTransferwise replied:
Done

int-chaos reviewed (commented):
Models and new features, like PCA and fourier transformations, look good to me.

markharley committed:
Merge branch 'main' into extract-task-class-from-automl

EgorKraevTransferwise committed:
Hotwire TS_Sklearn wrapper to fix test fail by setting
optimizer_for_horizon == False

markharley committed:
Revert changes to the automl_classification and pin FLAML version

markharley committed:
Merge branch 'extract-task-class-from-automl' of github.com:markharley/FLAML into extract-task-class-from-automl

markharley committed:
Fix imports in reverted notebook

markharley committed:
Fix FLAML version in automl notebooks

markharley committed:
Merge remote-tracking branch 'upstream/main' into extract-task-class-from-automl

markharley committed:
Fix ml.py line endings

sonichi committed:
Merge branch 'main' into extract-task-class-from-automl

markharley committed:
Merge remote-tracking branch 'upstream/main' into extract-task-class-from-automl

markharley committed:
Fix CLASSIFICATION task import in automl_classification notebook

markharley committed:
Merge branch 'extract-task-class-from-automl' of github.com:markharley/FLAML into extract-task-class-from-automl

markharley committed:
Uncomment pip install in notebook and revert import

Not convinced this will work because of installing an older version of
the package into the environment in which we're running the tests, but
let's see.

markharley committed:
Merge branch 'main' into extract-task-class-from-automl

markharley committed:
Revert c6a5dd1a0

markharley committed:
Merge branch 'extract-task-class-from-automl' of github.com:markharley/FLAML into extract-task-class-from-automl

markharley committed:
Merge branch 'extract-task-class-from-automl' into time-series-task

markharley committed:
Fix get_classification_objective import in suggest.py

markharley committed:
Remove hcrystallball docs reference in TS_Sklearn

skzhang1 committed:
Fix Dependabot alerts (#922)

* fix_alerts

* clean up

---------

Co-authored-by: “skzhang1” <“shaokunzhang529@gmail.com”>

Borda committed:
pre-commit: update config (#925)

* update config

* apply precommit

coffepowered committed:
Improve annotations in automl and ml modules (#919)

* begin annotation in automl.py and ml.py

* EstimatorSubclass + annotate metric

* review: fixes + setting fit_kwargs as proper Optional

* import from flaml.automl.model (import from flaml.model is deprecated)

* comment n_jobs in train_estimator as well

* better annotation in _compute_with_config_base

Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

---------

Co-authored-by: Andrea W <a.ruggerini@ammagamma.com>
Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

thinkall committed:
Updated dict type args default value to None (#927)

LittleLittleCloud committed:
fix #493 (#932)

* add custom meta feature function

* Update flaml/default/suggest.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

---------

Co-authored-by: littlelittlecloud <g2260578356@gmail.com>
Co-authored-by: Chi Wang <wang.chi@microsoft.com>

levscaut committed:
add PySparkOvertimeMonitor to avoid exceeding time budget (#923)

* merging

* clean commit

* Delete mylearner.py

This file is not needed.

* fix py4j import error

* more tolerant cancelling time

* fix problems following suggestions

* Update flaml/tune/spark/utils.py

Co-authored-by: Li Jiang <bnujli@gmail.com>

* remove redundant model

* Update test/spark/custom_mylearner.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* add docstr

* reverse change in gitignore

* Update test/spark/custom_mylearner.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

---------

Co-authored-by: Li Jiang <bnujli@gmail.com>
Co-authored-by: Chi Wang <wang.chi@microsoft.com>

sonichi committed:
minor change to notebook (#928)

EgorKraevTransferwise committed:
Merge branch 'extract-task-class-from-automl' into time-series-task

# Conflicts:
#	flaml/automl/automl.py
#	flaml/automl/model.py
#	flaml/automl/state.py
#	flaml/default/suggest.py

EgorKraevTransferwise committed:
Merge markharley:extract-task-class-from-automl into this

EgorKraevTransferwise committed:
Merge branch 'time-series-task' of https://github.com/markharley/FLAML into time-series-task

 Conflicts:
	flaml/automl/state.py
	flaml/default/suggest.py

EgorKraevTransferwise committed:
Fix import, remove smooth.py

Borda committed:
precommit: end-of-file-fixer (#929)

* precommit: end-of-file-fixer

* exclude .gitignore

* apply

---------

Co-authored-by: Shaokun <shaokunzhang529@gmail.com>

Borda committed:
adding precommit check (#930)

* adding precommit check

* run precommit

* Apply suggestions from code review

Co-authored-by: Zvi Baratz <z.baratz@gmail.com>

* apply precommit

---------

Co-authored-by: Zvi Baratz <z.baratz@gmail.com>

dependabot[bot] committed:
Bump http-cache-semantics from 4.1.0 to 4.1.1 in /website (#913)

Bumps [http-cache-semantics](https://github.com/kornelski/http-cache-semantics) from 4.1.0 to 4.1.1.
- [Release notes](https://github.com/kornelski/http-cache-semantics/releases)
- [Commits](https://github.com/kornelski/http-cache-semantics/compare/v4.1.0...v4.1.1)

---
updated-dependencies:
- dependency-name: http-cache-semantics
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shaokun <shaokunzhang529@gmail.com>
Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

EgorKraevTransferwise committed:
Fix dependencies to fix TFT fail on Windows Python 3.8 and 3.9

EgorKraevTransferwise committed:
Add tensorboardX dependency to fix TFT fail on Windows Python 3.8 and 3.9

EgorKraevTransferwise committed:
Set pytorch-lightning==1.9.0 to fix  TFT fail on Windows Python 3.8 and 3.9

EgorKraevTransferwise committed:
Set pytorch-lightning==1.9.0 to fix  TFT fail on Windows Python 3.8 and 3.9

EgorKraevTransferwise committed:
Disable PCA reduction of lagged features for now, to fix svd convervence fail

liususan091219 committed:
adding openai documentation (#938)

* adding openai documentation

* adding openai documentation

* Update website/docs/Examples/Integrate - OpenAI.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update website/docs/Examples/Integrate - OpenAI.md

Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

* Update website/docs/Examples/Integrate - OpenAI.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update website/docs/Examples/Integrate - OpenAI.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update website/docs/Examples/Integrate - OpenAI.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* addressing comments

* Update website/docs/Examples/Integrate - OpenAI.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update website/docs/Examples/Integrate - OpenAI.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* adding Azure

* rerun

* addressing comments about Azure

---------

Co-authored-by: Chi Wang <wang.chi@microsoft.com>
Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

sonichi committed:
improve max_valid_n and doc (#933)

* improve max_valid_n and doc

* Update README.md

Co-authored-by: Li Jiang <lijiang1@microsoft.com>

* newline at end of file

* doc

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>
Co-authored-by: Susan Xueqing Liu <liususan091219@users.noreply.github.com>
Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

skzhang1 committed:
Update doc and notebook (#946)

* update

* update doc

---------

Co-authored-by: “skzhang1” <“shaokunzhang529@gmail.com”>

sonichi committed:
ChatGPT support (#942)

* improve max_valid_n and doc

* Update README.md

Co-authored-by: Li Jiang <lijiang1@microsoft.com>

* add support for chatgpt

* notebook

* newline at end of file

* chatgpt notebook

* ChatGPT in Azure

* doc

* math

* warning, timeout, log file name

* handle import error

* doc update; default value

* paper

* doc

* docstr

* eval_func

* prompt and messages

* remove confusing words

* notebook name

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>
Co-authored-by: Susan Xueqing Liu <liususan091219@users.noreply.github.com>

markharley committed:
Extract task class from automl (#857)

* Refactor into automl subpackage

Moved some of the packages into an automl subpackage to tidy before the
task-based refactor. This is in response to discussions with the group
and a comment on the first task-based PR.

Only changes here are moving subpackages and modules into the new
automl, fixing imports to work with this structure and fixing some
dependencies in setup.py.

* Fix doc building post automl subpackage refactor

* Fix broken links in website post automl subpackage refactor

* Fix broken links in website post automl subpackage refactor

* Remove vw from test deps as this is breaking the build

* Move default back to the top-level

I'd moved this to automl as that's where it's used internally, but had
missed that this is actually part of the public interface so makes sense
to live where it was.

* Re-add top level modules with deprecation warnings

flaml.data, flaml.ml and flaml.model are re-added to the top level,
being re-exported from flaml.automl for backwards compatability. Adding
a deprecation warning so that we can have a planned removal later.

* Fix model.py line-endings

* WIP

* WIP - Notes below

Got to the point where the methods from AutoML are pulled to
GenericTask. Started removing private markers and removing the passing
of automl to these methods. Done with decide_split_type, started on
prepare_data. Need to do the others after

* Re-add generic_task

* Fix tests: add Task.__str__

* Fix tests: test for ray.ObjectRef

* Hotwire TS_Sklearn wrapper to fix test fail

* Remove unused data size field from Task

* Fix import for CLASSIFICATION in notebook

* Update flaml/automl/data.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Fix review comments

* Fix task -> str in custom learner constructor

* Remove unused CLASSIFICATION imports

* Hotwire TS_Sklearn wrapper to fix test fail by setting
optimizer_for_horizon == False

* Revert changes to the automl_classification and pin FLAML version

* Fix imports in reverted notebook

* Fix FLAML version in automl notebooks

* Fix ml.py line endings

* Fix CLASSIFICATION task import in automl_classification notebook

* Uncomment pip install in notebook and revert import

Not convinced this will work because of installing an older version of
the package into the environment in which we're running the tests, but
let's see.

* Revert c6a5dd1a0

* Revert "Revert c6a5dd1a0"

This reverts commit e55e35adea03993de87b23f092b14c6af623d487.

* Black format model.py

* Bump version to 1.1.2 in automl_xgboost

* Add docstrings to the Task ABC

* Fix import in custom_learner

* fix 'optimize_for_horizon' for ts_sklearn

* remove debugging print statements

* Check for is_forecast() before is_classification() in decide_split_type

* Attempt to fix formatting fail

* Another attempt to fix formatting fail

* And another attempt to fix formatting fail

* Add type annotations for task arg in signatures and docstrings

* Fix formatting

* Fix linting

---------

Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>
Co-authored-by: EgorKraevTransferwise <egor.kraev@transferwise.com>
Co-authored-by: Chi Wang <wang.chi@microsoft.com>
Co-authored-by: Kevin Chen <chenkevin.8787@gmail.com>

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/main' into time-series-task

# Conflicts:
#	flaml/automl/automl.py
#	flaml/automl/data.py
#	flaml/automl/ml.py
#	flaml/automl/model.py
#	flaml/automl/nlp/huggingface/utils.py
#	flaml/automl/state.py
#	flaml/automl/task/factory.py
#	flaml/automl/task/generic_task.py
#	flaml/automl/task/task.py
#	flaml/default/suggest.py
#	notebook/automl_xgboost.ipynb

liususan091219 committed:
fix delete (#950)

EgorKraevTransferwise committed:
Merge flaml/main into time_series_task

EgorKraevTransferwise committed:
Attempt to fix formatting

EgorKraevTransferwise committed:
Attempt to fix formatting

andreaw-ag committed:
tentatively implement holt-winters-no covariates

andreaw-ag committed:
fix forecast method, clean class

andreaw-ag committed:
checking external regressors too

andreaw-ag committed:
update test forecast

andreaw-ag committed:
remove duplicated test file, re-add sarimax, search space cleanup

coffepowered committed:
Update flaml/automl/model.py

removed links. Most important one probably was: https://robjhyndman.com/hyndsight/ets-regressors/

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

thinkall committed:
Support spark dataframe as input dataset and spark models as estimators (#934)

* add basic support to Spark dataframe

add support to SynapseML LightGBM model

update to pyspark>=3.2.0 to leverage pandas_on_Spark API

* clean code, add TODOs

* add sample_train_data for pyspark.pandas dataframe, fix bugs

* improve some functions, fix bugs

* fix dict change size during iteration

* update model predict

* update LightGBM model, update test

* update SynapseML LightGBM params

* update synapseML and tests

* update TODOs

* Added support to roc_auc for spark models

* Added support to score of spark estimator

* Added test for automl score of spark estimator

* Added cv support to pyspark.pandas dataframe

* Update test, fix bugs

* Added tests

* Updated docs, tests, added a notebook

* Fix bugs in non-spark env

* Fix bugs and improve tests

* Fix uninstall pyspark

* Fix tests error

* Fix java.lang.OutOfMemoryError: Java heap space

* Fix test_performance

* Update test_sparkml to test_0sparkml to use the expected spark conf

* Remove unnecessary widgets in notebook

* Fix iloc java.lang.StackOverflowError

* fix pre-commit

* Added params check for spark dataframes

* Refactor code for train_test_split to a function

* Update train_test_split_pyspark

* Refactor if-else, remove unnecessary code

* Remove y from predict, remove mem control from n_iter compute

* Update workflow

* Improve _split_pyspark

* Fix test failure of too short training time

* Fix typos, improve docstrings

* Fix index errors of pandas_on_spark, add spark loss metric

* Fix typo of ndcgAtK

* Update NDCG metrics and tests

* Remove unuseful logger

* Use cache and count to ensure consistent indexes

* refactor for merge maain

* fix errors of refactor

* Updated SparkLightGBMEstimator and cache

* Updated config2params

* Remove unused import

* Fix unknown parameters

* Update default_estimator_list

* Add unit tests for spark metrics

sonichi committed:
gpt-4 support; openai workflow fix; model str; timeout; voting (#958)

* workflow; model str; timeout

* voting

* notebook

* pull request

* recover workflow

* voted answer

* aoai

* ignore None answer

* default config

* note

* gpt-4

* n=5

* cleanup

* config name

* introduction

* readme

* avoid None

* add output/ to gitignore

* openai version

* invalid var

* comment long running cells

levscaut committed:
more tolerant time limit for test_overtime (#960)

* more tolerant time limit for test_overtime

* Cancel assertion becasue github VM sometimes is super slow

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>

coffepowered committed:
Merge branch 'main' into exp_smoothing

andreaw-ag committed:
prevent short series

andreaw-ag committed:
add docs

EgorKraevTransferwise committed:
Merge remote-tracking branch 'coffee/exp_smoothing' into time-series-task

# Conflicts:
#	flaml/automl/automl.py
#	flaml/automl/ml.py
#	flaml/automl/model.py
#	flaml/automl/state.py
#	flaml/automl/task/generic_task.py
#	test/automl/test_forecast.py

EgorKraevTransferwise committed:
First attempt at merging Holt-Winters

EgorKraevTransferwise committed:
Linter fix

EgorKraevTransferwise committed:
Add holt-winters to TimeSeriesTask.estimators

EgorKraevTransferwise committed:
Fix spark test fail

EgorKraevTransferwise committed:
Attempt to fix another spark test fail

EgorKraevTransferwise committed:
Attempt to fix another spark test fail

qingyun-wu committed:
Adding a test function for OpenAI completion in flaml  (#951)

* improve max_valid_n and doc

* Update README.md

Co-authored-by: Li Jiang <lijiang1@microsoft.com>

* add support for chatgpt

* notebook

* newline at end of file

* chatgpt notebook

* ChatGPT in Azure

* doc

* math

* warning, timeout, log file name

* handle import error

* doc update; default value

* paper

* doc

* docstr

* eval_func

* add a test func in completion

* update notebook

* update math notebook

* improve notebok

* lint and handle exception

* flake8

* exception in test

* add agg_method

* NameError

* refactor

* Update flaml/integrations/oai/completion.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update flaml/integrations/oai/completion.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* add example

* merge files from oai_eval_test

* Revert "merge files from oai_eval_test"
This reverts commit 1e6a550f913bb94df6e9680934ccb7175d00702e.

* merge

* save results to notebook_output

* update version and cache

* update doc

* save nb cell results to file

* fix typo in model name

* code improvements

* improve docstr

* docstr

* docstr on the Returns of test

---------

Co-authored-by: Chi Wang <wang.chi@microsoft.com>
Co-authored-by: Li Jiang <lijiang1@microsoft.com>
Co-authored-by: Susan Xueqing Liu <liususan091219@users.noreply.github.com>

AyeshGK committed:
Noctus contribute (#961)

* indentation space #884 changing indentation of python code blocks

* indentation space #884 changing indentation of python code blocks 2

thinkall committed:
add timeout to openai create (#973)

Borda committed:
ci: concurrency (#931)

* ci: concurrency

* no fail-fast

* Apply suggestions from code review

---------

Co-authored-by: Shaokun <shaokunzhang529@gmail.com>

dependabot[bot] committed:
Bump webpack from 5.74.0 to 5.76.1 in /website (#952)

Bumps [webpack](https://github.com/webpack/webpack) from 5.74.0 to 5.76.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.74.0...v5.76.1)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shaokun <shaokunzhang529@gmail.com>

dependabot[bot] committed:
Bump @sideway/formula from 3.0.0 to 3.0.1 in /website (#963)

Bumps [@sideway/formula](https://github.com/sideway/formula) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/sideway/formula/releases)
- [Commits](https://github.com/sideway/formula/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: "@sideway/formula"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shaokun <shaokunzhang529@gmail.com>

dependabot[bot] committed:
Bump eta from 1.12.3 to 2.0.0 in /website (#941)

Bumps [eta](https://github.com/eta-dev/eta) from 1.12.3 to 2.0.0.
- [Release notes](https://github.com/eta-dev/eta/releases)
- [Commits](https://github.com/eta-dev/eta/compare/v1.12.3...v2.0.0)

---
updated-dependencies:
- dependency-name: eta
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shaokun <shaokunzhang529@gmail.com>

coffepowered committed:
Add Holt-Winters exponential smoothing (#962)

* tentatively implement holt-winters-no covariates

* fix forecast method, clean class

* checking external regressors too

* update test forecast

* remove duplicated test file, re-add sarimax, search space cleanup

* Update flaml/automl/model.py

removed links. Most important one probably was: https://robjhyndman.com/hyndsight/ets-regressors/

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* prevent short series

* add docs

---------

Co-authored-by: Andrea W <a.ruggerini@ammagamma.com>
Co-authored-by: Chi Wang <wang.chi@microsoft.com>

sonichi committed:
autogen subpackage (#968)

* math utils in autogen

* cleanup

* code utils

* remove check function from code response

* comment out test

* GPT-4

* increase request timeout

* name

* logging and error handling

* better doc

* doc

* codegen optimized

* GPT series

* text

* no demo example

* math

* import openai

* import openai

* azure model name

* azure model name

* openai version

* generate assertion if necessary

* condition to generate assertions

* init region key

* rename

* comments about budget

* prompt

---------

Co-authored-by: Susan Xueqing Liu <liususan091219@users.noreply.github.com>

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/main' into time-series-task

# Conflicts:
#	flaml/automl/ml.py
#	flaml/automl/model.py
#	flaml/automl/task/generic_task.py
#	test/automl/test_forecast.py

liususan091219 committed:
handling nlp divide by zero (#926)

* handling nlp divide by zero

* catching zerodivisionerror

* catching zerodivisionerror

* catching zerodivisionerror

* addressing comments

* addressing comments

* updating test case

* update

* add blank to last line

* update nlp notebook

* rerun

* rerun

* sync with main

* add model selection for nlg

* addressing keyerror

* add raise exception

* update

* fix bug

* revert

* updating automl_nlp

* Update flaml/automl/model.py

Co-authored-by: Zvi Baratz <z.baratz@gmail.com>

* address comments

* address comments

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>
Co-authored-by: Zvi Baratz <z.baratz@gmail.com>

EgorKraevTransferwise committed:
Merge branch 'microsoft:main' into time-series-task

Borda committed:
set black with 120 line length (#975)

* set black with 120 line length

* apply pre-commit

* apply black

sonichi committed:
Post release update (#985)

* news update

* doc update

* avoid KeyError

* bump version to 1.2.1

* handle empty responses

* typo

* eval function

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/main' into time-series-task

# Conflicts:
#	flaml/automl/ml.py
#	flaml/automl/model.py
#	flaml/automl/state.py
#	flaml/automl/task/generic_task.py
#	flaml/automl/task/task.py
#	test/automl/test_forecast.py

EgorKraevTransferwise committed:
Merge remote-tracking branch 'mark_fork/time-series-task' into time-series-task

EgorKraevTransferwise committed:
Change Black max line length to 127

EgorKraevTransferwise committed:
Change Black max line length to 120

thinkall committed:
fix tests failure caused by version incompatibility (#995)

JaneIllario committed:
Expose feature and label transformer in automl.py (#993)

* expose label and feature transformer

* linter apply

* avoid undefined attribute in flaml/automl/automl.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* avoid undefined attribute in flaml/automl/automl.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* retrigger checks

* retrigger checks

---------

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

sonichi committed:
make context optional; improve error handling and doc (#997)

* make context optional

* better error handling and doc

* skip instantiation if no context

* skip test

liususan091219 committed:
update nlp notebook (#940)

* update nlp notebook

* rerun

* rerun

* removing redundant in notebook

* remove redundant content in nlp notebook

* update notebook

* update plot

* update plot

* update plot

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>

Commit:
Merge branch 'main' of https://github.com/microsoft/FLAML into time-series-task

EgorKraevTransferwise committed:
Add logging for ARIMA params, clean up time series models inheritance

EgorKraevTransferwise committed:
Add more logging for missing ARIMA params

EgorKraevTransferwise committed:
Remove a meaningless test causing a fail, add stricter check on ARIMA params

EgorKraevTransferwise committed:
Fix a bug in HoltWinters

EgorKraevTransferwise committed:
A pointless change to hopefully trigger the on and off KeyError in ARIMA.fit()

sonichi committed:
Blog post for LLM tuning (#986)

* outline

* revision

* eval function signature

* first draft

* link

* format

* example

* cleanup

* average

* move figure

* tldr

* bold

* bold

* tag

liususan091219 committed:
fix zerodivision (#1000)

* fix zerodivision

* update

* remove final

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>

sonichi committed:
extract code from text; solve_problem; request_timeout in config; improve code (#999)

* extract code from text

* solve_problem; request_timeout in config

* improve

* move import statement

* improve code

* generate assertions

* constant

* configs for implement; voting

* doc

* execute code in docker

* success indicator of code executation in docker

* success indicator

* execute code

* strip n

* add cost in generate_code

* add docstr

* filename

* bytes

* check docker version

* print log

* python test

* remove api key address

* rename exit code

* success exit code

* datasets

* exit code

* recover openai tests

* cache and pattern match

* wait

* wait

* cache and test

* timeout test

* python image name and skip macos

* windows image

* docker images

* volume path and yaml

* win path -> posix

* extensions

* path

* path

* path

* path

* path

* path

* path

* path

* path

* path

* path

* skip windows

* path

* timeout in windows

* use_docker

* use_docker

* hot fix from #1000

---------

Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

markharley committed:
Fix formatting

EgorKraevTransferwise committed:
Merge branch 'main' into time-series-task

sonichi committed:
version update post release v1.2.2 (#1005)

royninja committed:
fixing the typo #990 (#994)

* fixing the typo #990

* Update website/docs/Use-Cases/Auto-Generation.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* removing extra space : Update website/docs/Use-Cases/Auto-Generation.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update website/docs/Use-Cases/Auto-Generation.md

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update website/docs/Use-Cases/Auto-Generation.md

---------

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

luckyklyist committed:
fixed sentence misplace #998 (#1010)

Borda committed:
pyproject.toml & switch to Ruff (#976)

* unify config to pyproject.toml
replace flake8 with Ruff

* drop configs

* update

* fixing

* Apply suggestions from code review

Co-authored-by: Zvi Baratz <z.baratz@gmail.com>

* setup

* ci

* pr template

* reword

---------

Co-authored-by: Zvi Baratz <z.baratz@gmail.com>
Co-authored-by: Li Jiang <lijiang1@microsoft.com>

qingyun-wu committed:
update readme (#1014)

Co-authored-by: Li Jiang <lijiang1@microsoft.com>

sonichi committed:
raise content_filter error (#1018)

* raise content_filter error

* import error handling

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/main' into time-series-task

# Conflicts:
#	.flake8
#	flaml/automl/model.py

EgorKraevTransferwise committed:
Attempt to fix formatting

EgorKraevTransferwise committed:
Attempt to fix formatting

EgorKraevTransferwise committed:
Attempt to fix formatting

thinkall committed:
Fix catboost failure in mac-os python<3.9 (#1020)

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/main' into time-series-task

sonichi committed:
coding agent; logging (#1011)

* coding agent

* tsp

* tsp

* aoai

* logging

* compact

* Handle Import Error

* cost function

* reset counter; doc

* reset_counter

* home page update

* use case

* catboost in linux

* catboost

* catboost

* catboost

* doc

* intro

* catboost

thinkall committed:
Merge branch 'main' into time-series-task

garar committed:
Add mlflow_logging param (#1015)

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

liususan091219 committed:
fix NLP zero division error (#1009)

* fix NLP zero division error

* set predictions to None

* set predictions to None

* set predictions to None

* refactor

* refactor

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>
Co-authored-by: Chi Wang <wang.chi@microsoft.com>
Co-authored-by: Li Jiang <bnujli@gmail.com>

thinkall committed:
Merge branch 'main' into time-series-task

thinkall committed:
update max_spark_parallelism to fit in auto-scale spark cluster (#1008)

* update max_spark_parallelism to fit in auto-scale spark cluster

* update test

thinkall committed:
update spark session in spark tests (#1006)

* add mlflow and spark integration tests

* remove unused params

* remove mlflow tests

EgorKraevTransferwise committed:
Merge branch 'main' into time-series-task

thinkall committed:
Merge branch 'main' into time-series-task

sonichi committed:
Mark experimental classes; doc; multi-config trial (#1021)

* Mark experimental classes

* template

* multi model

* test

* multi-config doc

* doc

* doc

* test

---------

Co-authored-by: Li Jiang <bnujli@gmail.com>

BeibinLi committed:
Catch AuthenticationError trying different configs (#1023)

* Catch AuthenticationError trying different configs
While trying different openai `config_list`, some
configs might be outdated (e.g., an API key is expired).
In these cases, we don't want the program to crash.
Instead, we might want to try other configs.

* Lint whitespace

sonichi committed:
chat completion check (#1024)

* chat completion check

* add test

* doc

* timeout

* bump version to 1.2.4

liususan091219 committed:
update model of text summarization (#1030)

qingyun-wu committed:
Human agent (#1025)

* add human agent and chat agent

* feedback msg

* clean print

* remove redundant import

* make coding agent work

* import check

* terminate condition

* rename

* add docstr

* exitcode to str

* print

* save and execute code

* add max_turn_num

* add max_turn_num in test_agent.py

* reduce max_turn_num in the test

* change max_turn_num to max_consecutive_auto_reply

* update human proxy agent

* remove execution agent and dated docstr

* clean doc

* add back work_dir

* add is_termination_msg when mode is NEVER

* revise stop condition

* remove work_dir in coding agent

* human_proxy_agent docstr

* auto_reply

* clean auto_reply

---------

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

sonichi committed:
fix of website link (#1042)

qingyun-wu committed:
Blogpost (#1026)

* add 1m milestone blogpost

* format issues

* update subsection title

* acknowledgement

* Update website/blog/2023-05-07-1M-milestone/index.mdx

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update website/blog/2023-05-07-1M-milestone/index.mdx

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* update blogpost

* collaborators

* wording

* Azure Data to Azure Synapse

* name

* Azure Synapse Analytics

* tasks and search space

* Update website/blog/2023-05-07-1M-milestone/index.mdx

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

---------

Co-authored-by: Chi Wang <wang.chi@microsoft.com>
Co-authored-by: Li Jiang <bnujli@gmail.com>

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/main' into time-series-task

# Conflicts:
#	setup.py

EgorKraevTransferwise committed:
Merge branch 'time-series-task' of https://github.com/markharley/FLAML into time-series-task

EgorKraevTransferwise committed:
Attempt to fix formatting

EgorKraevTransferwise committed:
Add type annotations to _train_with_config() in state.py

Kyoshiin committed:
updated search space (#1044)

Co-authored-by: Pratyay Roy <63900765+pratyay-roy@users.noreply.github.com>

EgorKraevTransferwise committed:
Add type annotations to prepare_sample_train_data() in state.py

EgorKraevTransferwise committed:
Add docstring for time_col argument of AutoML.fit()

thinkall committed:
Fix PULL_REQUEST_TEMPLATE and improve test by removing unnecessary environment variable (#1043)

* Improve test by removing unnecessary environment variable

* Fix PULL_REQUEST_TEMPLATE

* Hide pre-commit check

* remove the checkbox for pre-commit

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

---------

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

sonichi committed:
response filter (#1039)

* response filter

* rewrite implement based on the filter

* multi responses

* abs path

* code handling

* option to not use docker

* context

* eval_only -> raise_error

* notebook

* utils

* utils

* separate tests

* test

* test

* test

* test

* test

* test

* test

* test

* **config in test()

* test

* test

* filename

EgorKraevTransferwise committed:
Merge branch 'main' into time-series-task

sonichi committed:
Blogpost for adaptation in HumanEval (#1048)

* Blogpost for adaptation in HumanEval

* doc

* fix link

* fix link

* explain

* model

* interface

* link

* typo

* doc

EgorKraevTransferwise committed:
Address @sonichi's comments on PR

EgorKraevTransferwise committed:
Merge branch 'time-series-task' of https://github.com/markharley/FLAML into time-series-task

EgorKraevTransferwise committed:
Fix formatting

EgorKraevTransferwise committed:
Fix formatting

EgorKraevTransferwise committed:
Merge branch 'main' into time-series-task

EgorKraevTransferwise committed:
Reduce test time budget

EgorKraevTransferwise committed:
Reduce test time budget

EgorKraevTransferwise committed:
Merge branch 'time-series-task' of https://github.com/markharley/FLAML into time-series-task

EgorKraevTransferwise committed:
Increase time budget for the test to pass

sonichi committed:
Improve messaging in documentation (#1050)

* Improve messaging in documentation

* doc

* improve wording in blogpost

---------

Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

sonichi committed:
create an automl option to remove unnecessary dependency for autogen and tune (#1007)

* version update post release v1.2.2

* automl option

* import pandas

* remove automl.utils

* default

* test

* type hint and version update

* dependency update

* link to open in colab

* use packging.version to close #725

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>
Co-authored-by: Li Jiang <bnujli@gmail.com>

badjouras committed:
docs: 📝 Fix link to installation section in Task-Oriented-AutoML.md (#1051)

Co-authored-by: rcardoso <ricardo.cardoso@anova.com>
Co-authored-by: Chi Wang <wang.chi@microsoft.com>

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/main' into time-series-task

# Conflicts:
#	flaml/automl/ml.py
#	flaml/automl/model.py
#	flaml/automl/state.py
#	flaml/automl/task/factory.py
#	flaml/default/suggest.py

EgorKraevTransferwise committed:
Remove redundant imports

EgorKraevTransferwise committed:
Remove more redundant imports

EgorKraevTransferwise committed:
Minor fixes of points raised by Qingyun

sonichi committed:
doc and test update (#1053)

* doc and test update

* docker update

qingyun-wu committed:
remove redundant doc and add tutorial (#1004)

* remove redundant doc and add tutorial

* add demos for pydata2023

* Update pydata23 docs

* remove redundant notebooks

* Move tutorial notebooks to notebook folder

* update readme and notebook links

* update notebook links

* update links

* update readme

---------

Co-authored-by: Li Jiang <lijiang1@microsoft.com>
Co-authored-by: Li Jiang <bnujli@gmail.com>

EgorKraevTransferwise committed:
Try to fix pandas import fail

qingyun-wu committed:
add agent notebook and documentation (#1052)

* add agent notebook and documentation

* fix bug

* set flush to True when printing msg in agent

* add a math problem in agent notebook

* remove

* header

* improve notebook doc

* notebook update

* improve notebook example

* improve doc

* improve notebook doc

* improve print

* doc

* human_input_mode

* human_input_mode str

* indent

* indent

* Update flaml/autogen/agent/user_proxy_agent.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update notebook/autogen_agent.ipynb

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update notebook/autogen_agent.ipynb

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update notebook/autogen_agent.ipynb

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* renaming and doc format

* typo

---------

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Try to fix pandas import fail, again

EgorKraevTransferwise committed:
Formatting fixes

EgorKraevTransferwise committed:
More formatting fixes

EgorKraevTransferwise committed:
Added test that loops over TS models to ensure coverage

EgorKraevTransferwise committed:
Fix formatting issues

EgorKraevTransferwise committed:
Fix more formatting issues

EgorKraevTransferwise committed:
Merge remote-tracking branch 'origin/main' into time-series-task

EgorKraevTransferwise committed:
Fix random fail in check

thinkall committed:
Support more azure openai api_type (#1059)

thinkall committed:
suppress warning message of pandas_on_spark to_spark (#1058)

EgorKraevTransferwise committed:
Put back in tests for ARIMA predict without fit

EgorKraevTransferwise committed:
Merge branch 'main' into time-series-task

sonichi committed:
Agent notebook example with human feedback; Support shell command and multiple code blocks; Improve the system message for assistant agent; Improve utility functions for config lists; reuse docker image  (#1056)

* add agent notebook and documentation

* fix bug

* set flush to True when printing msg in agent

* add a math problem in agent notebook

* remove

* header

* improve notebook doc

* notebook update

* improve notebook example

* improve doc

* agent notebook example with user feedback

* log

* log

* improve notebook doc

* improve print

* doc

* human_input_mode

* human_input_mode str

* indent

* indent

* Update flaml/autogen/agent/user_proxy_agent.py

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* shell command and multiple code blocks

* Update notebook/autogen_agent.ipynb

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update notebook/autogen_agent.ipynb

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* Update notebook/autogen_agent.ipynb

Co-authored-by: Chi Wang <wang.chi@microsoft.com>

* coding agent

* math notebook

* renaming and doc format

* typo

* infer lang

* sh

* docker

* docker

* reset consecutive autoreply counter

* fix explanation

* paper talk

* human feedback

* web info

* rename test

* config list explanation

* link to blogpost

* installation

* homepage features

* features

* features

* rename agent

* remove notebook

* notebook test

* docker command

* notebook update

* lang -> cmd

* notebook

* make it work for gpt-3.5

* return full log

* quote

* docker

* docker

* docker

* docker

* docker

* docker image list

* notebook

* notebook

* use_docker

* use_docker

* use_docker

* doc

* agent

* doc

* abs path

* pandas

* docker

* reuse docker image

* context window

* news

* print format

* pyspark version in py3.8

* pyspark in py3.8

* pyspark and ray

* quote

* pyspark

* pyspark

* pyspark

---------

Co-authored-by: Qingyun Wu <qingyun.wu@psu.edu>

sonichi committed:
Fix documentation (#1075)

* Fix indentation in documentation

* newline

* version

sonichi committed:
encode timeout msg in bytes (#1078)

* encode timeout msg in bytes

* fix msg and test

qingyun-wu committed:
add pands requirement in benchmark option (#1070)

Co-authored-by: Li Jiang <bnujli@gmail.com>

EgorKraevTransferwise committed:
Put back in tests for lgbm

EgorKraevTransferwise committed:
Merge branch 'time-series-task' of https://github.com/markharley/FLAML into time-series-task

EgorKraevTransferwise committed:
Merge branch 'main' into time-series-task

sonichi committed:
Update test/test_model.py

cover dedup

EgorKraevTransferwise committed:
Match target length to X length in missing test
